### ISSUE TITLE
PolymerBase.arrayDelete does not notify when Array argument is passed

### ISSUE THREAD

**Issue by jklein24** [Polymer/polymer#2427]
It looks like arrayDelete uses path.splice instead of this.splice if an array argument is passed rather than a string argument.

https://github.com/Polymer/polymer/blob/master/src/standard/utils.html#L214

This means that the array is modified, but no observers are called.

**Comment by arthurevans** [comment]
I think this is probably "expected" in the sense that we can't notify unless we have a path. But it is confusing to the non-initiate. And the initiate, for that matter. So it should probably be noted.

**Comment by kevinpschaaf** [comment]
@arthurevans That's right, it's intentional.  `arrayDelete` (taking an array reference) pre-dated the array mutation API's, and so the previous behavior was left as a fallback for backwards-compatibility.  Thanks for updating the documentation; I'll go ahead and merge that.

**Comment by beckula** [comment]
Hi guys.  So I consider myself an "initiate" by now, and today I passed an array into this.arrayDelete and scratched my head wondering why the UI didn't update.  Then I remembered this thread, but when I checked the API page (http://polymer.github.io/polymer/) it still doesn't mention anything about this.  It would be nice if it made it clear about the array not triggering a notification.  Someone out there is probably scratching their head right now....or me again in 5 months.  :)
